batch-jberet/src/test/java/org/wildfly/extension/batch/jberet/JBeretSubsystemParsingTestCase.java

```diff
@@ -151,6 +164,42 @@ public FileVisitResult visitFile(final Path file, final BasicFileAttributes attr
         }
     }
 
+    @Test
+    public void testRejectingTransformersEAP74() throws Exception {
+        FailedOperationTransformationConfig transformationConfig = new FailedOperationTransformationConfig();
+
+        PathAddress repositoryAddress = PathAddress.pathAddress(BatchSubsystemDefinition.SUBSYSTEM_PATH, InMemoryJobRepositoryDefinition.PATH);
+        transformationConfig.addFailedAttribute(repositoryAddress,
+                new FailedOperationTransformationConfig.NewAttributesConfig(CommonAttributes.EXECUTION_RECORDS_LIMIT));
+
+        PathAddress jdbcRepositoryAddress = PathAddress.pathAddress(BatchSubsystemDefinition.SUBSYSTEM_PATH, JdbcJobRepositoryDefinition.PATH);
+        transformationConfig.addFailedAttribute(jdbcRepositoryAddress,
+                new FailedOperationTransformationConfig.NewAttributesConfig(CommonAttributes.EXECUTION_RECORDS_LIMIT));
+
+        testRejectingTransformers(transformationConfig, ModelTestControllerVersion.EAP_7_4_0);
+    }
+
+    private void testRejectingTransformers(FailedOperationTransformationConfig transformationConfig, ModelTestControllerVersion controllerVersion) throws Exception {
+        ModelVersion subsystemModelVersion = controllerVersion.getSubsystemModelVersion(BatchSubsystemDefinition.NAME);
+
+        KernelServicesBuilder builder = createKernelServicesBuilder(createAdditionalInitialization());
+        LegacyKernelServicesInitializer kernelServicesInitializer = builder.createLegacyKernelServicesBuilder(createAdditionalInitialization(), controllerVersion, subsystemModelVersion)
+                .addMavenResourceURL("org.wildfly.core:wildfly-threads:" + controllerVersion.getCoreVersion())
+                .dontPersistXml();
+        try {
+            BatchSubsystemExtension.class.getClassLoader().loadClass("javax" + ".batch.operations.JobStartException");
+            kernelServicesInitializer.addMavenResourceURL("org.jboss.eap:wildfly-batch-jberet:" + controllerVersion.getMavenGavVersion());
+        } catch (ClassNotFoundException e) {
+            kernelServicesInitializer.addMavenResourceURL("org.wildfly:wildfly-batch-jberet-jakarta:26.0.0.Final");
+        }
```

Comment:
I'm bit unsure about this piece.

The batch subsystem module (and this test class with it) is transformed into Jakarta-version subsystem with renamed package names (wildfly-batch-jberet -> wildfly-batch-jberet-jakarta).

In the original subsystem, I need to do 

```kernelServicesInitializer.addMavenResourceURL("org.jboss.eap:wildfly-batch-jberet:" + controllerVersion.getMavenGavVersion())```

when testing that the config can be transformed to previous EAP version.

This stops working when the test is run in the transformed "-jakarta" module, because the `org.jboss.eap:wildfly-batch-jberet` code would still use javax API, while it needs to use jakarta API.

So when this test is run in the transformed module, I need to include jakarta version of the subsystem instead:
```
kernelServicesInitializer.addMavenResourceURL("org.wildfly:wildfly-batch-jberet-jakarta:26.0.0.Final");
```
As JBoss EAP doesn't productize the jakarta version modules, I'm using "org.wildfly" groupId here.

Also, the only way that occurs to me how to detect that the test is run in "-jakarta" module is to try to load some javax.batch API class, and if that fails, I'm in the jakarta world.

Does that look alright? Is there better way to do this?

Reply:
@TomasHofman what you have is ok; another way is to do 

if (JobStartException.class.getName().startsWith("jakarta.") { // i'm in jakarta land }

If this is working I'd say leave it as is though.

Reply:
@bstansberry thanks for confirming! I'm keeping the current version then :+1: